CMakeLists.txt

```diff
@@ -56,6 +56,30 @@ set(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE}
     CACHE STRING "Choose the type of build: None Debug Release Coverage ASan ASanDbg MemSan MemSanDbg Check CheckFull"
     FORCE)
 
+# Create a symbolic link from ${base_name} in the binary directory
+# to the corresponding path in the source directory.
+function(link_to_source base_name)
+    # Get OS dependent path to use in `execute_process`
+    file(TO_NATIVE_PATH "${CMAKE_CURRENT_BINARY_DIR}/${base_name}" link)
+    file(TO_NATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${base_name}" target)
+
+    if (NOT EXISTS ${link})
+        if (CMAKE_HOST_UNIX)
+            set(command ln -s ${target} ${link})
+        else()
+            set(command cmd.exe /c mklink /j ${link} ${target})
```

Comment:
`mklink` consistently fails for me on my test installation because it doesn't like my VM.

Given we're creating links to data files for the sake of executing tests, I'd propose it makes sense to have as an option the ability to make a local copy of those files for the sake of tests. 

That idea may be better off in it's own PR, but I thought I'd mention it while you're in the area.

Reply:
You mean add a way to do a copy instead of a link? I did wonder if this actually worked on all supported Windows build environments, but given my Windows and cmake inexperience, I don't think I'm the best person for that. So I'd rather handle this (which is a preexisting issue) separately.

Reply:
I have raised this as issue #1496. Not very urgent, but I do see it as more of a bug than a feature request.